hy/core/shadow.hy

```diff
@@ -61,4 +61,32 @@
         (reduce operator.truediv args)))))
 
 
-(setv *exports* ['+ '- '* '/])
+(defn comp-op [op args]
+  "Helper for shadow comparison operators"
+  (if (< (len args) 2)
+    (raise (TypeError "Need at least 2 arguments to compare"))
+    (reduce operator.and_
+            (list-comp (op x y)
+                       [(, x y) (zip args (slice args 1))]))))
+(defn < [&rest args]
+  "Shadow < operator for when we need to import / map it against something"
+  (comp-op operator.lt args))
+(defn <= [&rest args]
+  "Shadow <= operator for when we need to import / map it against something"
+  (comp-op operator.le args))
+(defn = [&rest args]
+  "Shadow = operator for when we need to import / map it against something"
+  (comp-op operator.eq args))
+(defn != [&rest args]
+  "Shadow != operator for when we need to import / map it against something"
+  (comp-op operator.ne args))
+(defn >= [&rest args]
+  "Shadow >= operator for when we need to import / map it against something"
+  (comp-op operator.ge args))
+(defn > [&rest args]
+  "Shadow > operator for when we need to import / map it against something"
+  (comp-op operator.gt args))
+; TODO figure out a way to shadow "is" (and then shadow "is_not" too)
```

Comment:
They could be, but not the keyword version… you’d need to import operators.and_

> On Mar 12, 2015, at 12:40 AM, Tianon Gravi <notifications@github.com> wrote:
> 
> In hy/core/shadow.hy:
> 
> > +(defn <= [&rest args]
> > +  "Shadow <= operator for when we need to import / map it against something"
> > +  (comp-op operator.le args))
> > +(defn = [&rest args]
> > +  "Shadow = operator for when we need to import / map it against something"
> > +  (comp-op operator.eq args))
> > +(defn != [&rest args]
> > +  "Shadow != operator for when we need to import / map it against something"
> > +  (comp-op operator.ne args))
> > +(defn >= [&rest args]
> > +  "Shadow >= operator for when we need to import / map it against something"
> > +  (comp-op operator.ge args))
> > +(defn > [&rest args]
> > +  "Shadow > operator for when we need to import / map it against something"
> > +  (comp-op operator.gt args))
> > +; TODO figure out a way to shadow "is" (and then shadow "is_not" too)
> 
> I suppose this would apply to and and or as well, since I don't believe and can be shadowed either.
> 
> —
> Reply to this email directly or view it on GitHub.
>